internal/process/defunct_processes.go

```diff
@@ -16,50 +16,60 @@ const ProcessFS = "/proc"
 
 // Stat represents status information of a process from /proc/[pid]/stat.
 type Stat struct {
+	// Pid is the PID of the process
+	Pid int
+
 	// Comm is the command name (usually the executable filename).
 	Comm string
 
 	// State is the state of the process.
 	State string
+
+	// PPid is the parent PID of the process
+	PPid int
 }
 
-// DefunctProcesses returns the number of zombie processes in the node.
-func DefunctProcesses() (defunctCount uint, retErr error) {
-	return DefunctProcessesForPath(ProcessFS)
+// ParseDefunctProcesses returns the number of defunct processes on the node,
+// as well as the number of defunct children of the current running process.
+func ParseDefunctProcesses() (defunctCount uint, defunctChildren []int, retErr error) {
+	return ParseDefunctProcessesForPathAndParent(ProcessFS, os.Getpid())
 }
 
-// DefunctProcessesForPath retrieves the number of zombie processes from
-// a specific process filesystem.
-func DefunctProcessesForPath(path string) (defunctCount uint, retErr error) {
+// ParseDefunctProcessesForPath retrieves the number of zombie processes from
```

Comment:
Is this function public only because we have to test it, and the test package is named `..._test`?

If yes, we're increasing the API surface (even if it's internal) to merely test it, and this does seem like a good practice to me .

Reply:
correct, and it is (perhaps unfortunately) idiomatic in the cri-o repo. I think I would leave it as is for now

Reply:
So, essentially we do two things
1. Use `<package>_test` in tests in order to have "black box" testing (i.e. only test public APIs).
2. Make internal functions we want to test public.

Seems like 2 defeats the purpose of 1 and actually make things worse.

I totally agree this is not to be addressed in this PR, but overall this seems like an issue we'd want to address.

Reply:
one thing worth noting Is since this library is in `internal/` directory it can't actually be imported publically. that does confuse the interfaces in cri-o, but at least we're the only ones who have to resist the temptation to call a function that is external but should be treated as internal

that said, this is an idiom that is consistent across cri-o, so some random person could import functions in `server` that aren't really meant for public use either..